packages/examples/kura/integration_with_kura.py

```diff
@@ -0,0 +1,257 @@
+delay_retries = random.randint(lfm(delay_retries_min), lfm(delay_retries_max))
+
+lock = threading.Lock()
+Msg_dic = {}                #{"demo": {"flag": 0, "msg": 0, "metric": metric}}
+Gatewaydict = {}            #{"assetName": reg_device}
+
+callbackFunc_str = '''def metric_callback_%s():    return Msg_dic['%s']['msg']''' # update the data by use Msg_dic disc
+
+"""Subscibe callback function"""
+def sub_callback(self, client, userdata, msg):
+
+    if (msg.topic.split("/").__len__() == 4) and (msg.topic.split("/")[3] == "pulse-topic"): # msg.topic = "pulse-account-name/pulseid/pulse/pulse-topic"
+
+        Jsonpayload = json.loads(msg.payload)['metrics']
+        if Jsonpayload: # msg.payload = "{"sentOn":1533007387525,"metrics":{"assetName":"Store-GW-03","sensor_timestamp":1533007387525,"sensor":0}}"
+
+            #is new gateway?
+            if Jsonpayload['assetName'] in Gatewaydict.keys():
+
+                log.info("is old gateway")
+
+                for key_metric in Jsonpayload.keys():
+
+                    # filter the unuseful sensor
+                    if key_metric == "assetName":
+                        continue
+                    if key_metric.find('stamp') >= 0:
+                        continue
+
+                    if key_metric in Msg_dic.keys(): #exist update msg
+
+                        Msg_dic[key_metric]['msg'] = Jsonpayload[key_metric]
+                        Msg_dic[key_metric]['flag'] = 0
+
+                    else:
+
+                        Msg_dic[key_metric] = {}
+                        Msg_dic[key_metric]['msg'] = Jsonpayload[key_metric]
+                        Msg_dic[key_metric]['flag'] = 1
+
+                        exec(callbackFunc_str % (key_metric, key_metric)) # Dynamically generating callback function
+
+                        try:
+                            metric_ins = Metric(name=key_metric,
+                                                unit=None, interval=3,
+                                                aggregation_size=1,
+                                                sampling_function=eval("metric_callback_%s" % key_metric) # Callback function publish data to iotcc
+                                                )
+                            reg_metric = userdata['self'].iotcc.register(metric_ins)
+                            userdata['self'].iotcc.create_relationship(Gatewaydict[Jsonpayload['assetName']], reg_metric)
+                            reg_metric.start_collecting()
+                            userdata['self'].metrics.append(reg_metric)
+                            Msg_dic[key_metric]['metric'] = reg_metric
+
+                        except Exception as e:
+                            log.error(
+                                'Exception while loading metric {0} for Edge System {1}'.format
+                                (key_metric, str(e)))
+
+
+            else: #new device enrolling
+
+                log.info("is new gateway")
```

Comment:
Change the log message to new Device to be registered.

Reply:
Fixed.